compiler/rustc_lint/messages.ftl

```diff
@@ -733,6 +733,8 @@ lint_reserved_prefix = prefix `{$prefix}` is unknown
     .label = unknown prefix
     .suggestion = insert whitespace here to avoid this being parsed as a prefix in Rust 2021
 
+lint_self_type_conversion = this conversion is useless `{$source}` to `{$target}`
```

Comment:
Incorrect sentence structure...

```suggestion
lint_self_type_conversion = useless conversion to the same type: `{$target}`
```

(taken from https://github.com/rust-lang/rust-clippy/blob/e5a1ef0795a119a68cbd4cccb41b569206d74d78/clippy_lints/src/useless_conversion.rs#L187)

Reply:
The lint itself will need to be iterated on. This is all a placeholder so that I can run crater :)